source/extensions/filters/common/fault/fault_config.cc

```diff
@@ -11,24 +11,54 @@ namespace Filters {
 namespace Common {
 namespace Fault {
 
+envoy::type::v3::FractionalPercent
+HeaderPercentageProvider::percentage(const Http::RequestHeaderMap* request_headers) const {
+  if (!request_headers) {
+    return percentage_;
+  }
+
+  const auto header = request_headers->get(header_name_);
+  if (header == nullptr) {
+    return percentage_;
+  }
+
+  uint64_t header_numerator;
+  if (!absl::SimpleAtoi(header->value().getStringView(), &header_numerator)) {
+    return percentage_;
+  }
+
+  if (header_numerator > 100) {
```

Comment:
Sorry just noticed this. Fractional percent supports 1/100, 1/10,000, etc. Do we want to support this somehow in a header? I'm wondering if this should somehow be scaled by the default percentage that is configured? WDYT?

Reply:
We could definitely use a denominator of a configured default percentage - that could work. If we do this we could also have a logic that would compute an output percentage's numerator as the minimum of two values - the value of HTTP percentage header and the value defined in default configured percentage. 

Why would this additional complexity be helpful?
- That would be an additional safety mechanism that would allow Envoy users to define the maximum percentage of requests a given type of fault should be applied to. 
- It would ensure that no change in the configuration of a default percentage results in too many faults being ingested via HTTP headers. Example when the lack of this mechanism could lead to unintended behavior is presented below.

### Example

#### Step 1
default percentage configuration:
```proto
numerator: 100
denumerator: 1000
```
Effective fault injection rate is 10% since 100/1000 = 10.

HTTP headers used by clients injecting faults via HTTP headers:
```proto
`x-fault-abort-request-abort`: 404
`x-fault-abort-request-abort-percentage`: 100
```
Effective fault injection rate is 10% since 100/1000 = 10.
#### Step 2
Somebody on the server side decides to change a default percentage configuration to:
```proto
numerator: 10
denumerator: 100
```
From their perspective there should be no negative results of this change since the effective failure rate doesn't change (it's still equal to 10% since 100/1000 = 10/100 = 10)
The person responsible for a change doesn't update clients that use HTTP headers to ingest faults.

##### What happens with clients using HTTP headers to inject faults?
###### Without proposed min logic for the calculation of a numerator
```proto
`x-fault-abort-request-abort`: 404
`x-fault-abort-request-abort-percentage`: 100
```
Effective fault injection rate increases to 100% since 100/100 (new denominator of a default percentage) = 100.
###### With proposed min logic for the calculation of a numerator
```proto
`x-fault-abort-request-abort`: 404
`x-fault-abort-request-abort-percentage`: 100
```
Effective fault injection rate is still at 100% since min(100/100, 10 [Default Percentage Effective Rate]) = 100.

Let me know what you think about the proposed additional to your proposed change.

Reply:
Thanks for the super detailed proposal! I agree the capping makes sense, but please make sure it's well documented.

Reply:
Thanks for your suggestion! Implemented here https://github.com/envoyproxy/envoy/pull/10724/files#diff-186e4ff5610d03e9dc6f8fcace2aa4a3R26-R29.